src/compiler/nvcc.rs

```diff
@@ -256,14 +274,34 @@ mod test {
     use std::collections::HashMap;
     use std::path::PathBuf;
 
-    fn parse_arguments_(arguments: Vec<String>) -> CompilerArguments<ParsedArguments> {
+    fn parse_arguments_gcc(arguments: Vec<String>) -> CompilerArguments<ParsedArguments> {
+        let arguments = arguments.iter().map(OsString::from).collect::<Vec<_>>();
+        Nvcc {
+            host_compiler: NvccHostCompiler::Gcc,
+            version: None,
+        }
+        .parse_arguments(&arguments, ".".as_ref())
+    }
+    fn parse_arguments_nvc(arguments: Vec<String>) -> CompilerArguments<ParsedArguments> {
         let arguments = arguments.iter().map(OsString::from).collect::<Vec<_>>();
-        Nvcc { version: None }.parse_arguments(&arguments, ".".as_ref())
+        Nvcc {
+            host_compiler: NvccHostCompiler::Nvhpc,
+            version: None,
+        }
+        .parse_arguments(&arguments, ".".as_ref())
     }
```

Comment:
Shouldnt there also be a `parse_arguments_msvc` function?

Reply:
Done